### PR TITLE
Enhancement/24 update templates

### LINKED ISSUE #24: Updates in the templates
1. New github workflow `maven_central_release.yml` (only for projects that require it): https://github.com/exasol/virtual-schema-common-java/blob/master/.github/workflows/maven_central_release.yml

2. Rename a workflow for dependencies check to `dependencies_check.yml`. Update its content: https://github.com/exasol/virtual-schemas/blob/master/.github/workflows/dependencies_check.yml

Additional pom file plugins for Maven Release:
3. Update in the `maven-gpg-plugin`
4. New plugin `maven-deploy-plugin`
5. New plugin `nexus-staging-maven-plugin`

See the plugin templates in this commit: https://github.com/exasol/virtual-schema-common-java/commit/65c76b9e098eced7d5f88b6e0f228de0a77fa37a

The plugin changes 3-5 should be released after the release robot is released because you probably can't release on MC manually anymore after you add those changes

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #24

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,11 +66,14 @@ This plugin provides different template modules for different kinds of projects.
...
-*Not yet implemented.*
 
-This module will check required configuration for releasing on maven central in the future.
-You can, however, already add this module. By that you can't forget to add it later.
+This module checks the required configuration for releasing on maven central.
 
+* required maven plugins
+    * [maven-gpg-plugin](https://maven.apache.org/plugins/maven-gpg-plugin/)
+    * [maven-deploy-plugin](https://maven.apache.org/plugins/maven-deploy-plugin/)
+    * [nexus-staging-maven-plugin](https://github.com/sonatype/nexus-maven-plugins/tree/master/staging/maven-plugin)
```
Comment (AnastasiiaSergienko) [inline README.md:75]: I think we're missing some other plugins here. I'm not sure if you check them in some other module, but the full list is: 
"nexus-staging-maven-plugin", "maven-source-plugin", "maven-gpg-plugin", "maven-javadoc-plugin", "maven-deploy-plugin".
I've also added a template with them to the release robot: https://github.com/exasol/release-robot/blob/feature/50_add_validations_for_maven_release/doc/user_guide/maven_central_release_pom_file_template.md

File: doc/changes/changes_0.3.0.md [hunk 2]
```diff
@@ -5,13 +5,22 @@
 ## Summary
 
 Added new validations:
+
 * doc/changes/changelog_X.X.X.md
 * travis.yml
 * LICENSE
+* for maven-central: .github/workflows/maven_central_release.yml
+* maven-gpg-plugin
+* maven-deploy-plugin
```
Comment (AnastasiiaSergienko) [inline doc/changes/changes_0.3.0.md:14]: + nexus-staging-maven-plugin

File: src/main/resources/maven_templates/maven-gpg-plugin.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    <version>1.6</version>
+    <executions>
+        <execution>
+            <id>sign-artifacts</id>
+            <phase>verify</phase>
+            <goals>
+                <goal>sign</goal>
+            </goals>
+            <configuration>
+                <!-- Prevent `gpg` from using pinentry programs -->
```
Comment (AnastasiiaSergienko) [inline src/main/resources/maven_templates/maven-gpg-plugin.xml:13]: We can remove this comment

**Review by AnastasiiaSergienko — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jakobbraun) [reply README.md:75]: Yes. This is just the plugins that were mentioned in the issue. The other plugins I will add when fixing issue #18

File: doc/changes/changes_0.3.0.md (hunk 2 above)
Comment (jakobbraun) [reply doc/changes/changes_0.3.0.md:14]:
```suggestion
* maven-deploy-plugin
* nexus-staging-maven-plugin
```

File: src/main/resources/maven_templates/maven-gpg-plugin.xml (hunk 3 above)
Comment (jakobbraun) [reply src/main/resources/maven_templates/maven-gpg-plugin.xml:13]:
```suggestion
```

File: README.md (hunk 1 above)
Comment (AnastasiiaSergienko) [reply README.md:75]: Okay

**Review by AnastasiiaSergienko — APPROVED** [review]

**Pull request merged**